### DISCUSSION TITLE
Labo Tetris 1 - question au niveau de pièces tombantes qui serait illégales avant la suppression de ligne mais pas après

### DISCUSSION THREAD

**Discussion by A-Grunder** [web-classroom/web-classroom.github.io#7] [Q&A : Pratique / Labos] [answered]
### Légende :
 - "1" une case d'une pièces tombante du joueur 1
 - "2" une case d'une pièce tombante du joueur 2
 - "." une case vide
 - "x" une case "mise à terre"
 - "a/b" un overlap entre "a" et "b"

par exemple dans le cas suivant :
```
.   .   2   2
.   .   2   2
.   1   1  . 
x   x   1   1


.   .   .   .
.   .   2   2
.   1  1/2  2 
x   x   1   1


.   .   .   .
.   .   2   2
.   x  x/2  2
x   x   x   x
``` 

est-ce que c'est :
```
.   .   .   .
.   .   2   2
.   .   2   2
.   x   x   .
``` 

ou :
```
.   .   .   .
.   .   .   .
.   .   .   .
.   x   x   .
```

**Comment by OlivierLmr** [comment, marked as answer]
Très bonne question, un cas limite que je n'avais pas pensé à tester ni à considérer dans l'énoncé.

En gros, il s'agit de donner un ordre entre déplacement, grounding, suppression de lignes et invalidation de pièces tombantes.

Puisque rien n'est précisé, vous êtes libres de faire comme vous voulez. Personnellement, je trouve que la suppression de ligne et le grounding vont un peu ensemble, et on aurait donc que l'invalidation a bien lieu après la suppression de ligne (donc la pièce 2 ne serait pas invalidée dans votre exemple). Mais si vous avez une autre préférence, faites seulement ; c'est un cas limite qui ne sera pas testé.
